src/modal.ts

```diff
@@ -0,0 +1,67 @@
+import {authWithProvider} from './actions'
+import {Web3AuthProviderMetadata} from './types'
+
+export const getButtonInnerHtml = (serviceName: string, imgSrc: string) => `
```

Comment:
for now this seems good enough. As a follow up, I'd consider using some framework to generate the html and manage possible dynamic updates of the elements and embed it into an iframe to prevent app css/DOM manipulation from interfering with our widget. I'd give a shot e.g. to preact being embedded like that - which I found this blog about: https://dev.to/companycam/build-an-embeddable-widget-using-preact-and-the-shadow-dom-33lm - what do you think? Did you consider anything similar?

Reply:
I created an issue for this #5